### PR TITLE
Configurator Vue mode update

### PR DESCRIPTION
**Pull request by HansJuice**
- Fixed caching issue;
- Static extra data is loaded through initial query instead of one query per step (this improves performance);
- Extra data from the extra data query can now overwrite static extra data values;
- Support for steps that open modal when clicking on option instead of making a selection.

Asana ticket for the caching issue fix: https://app.asana.com/0/1200923549887805/1204833966511650

### CODE REVIEW

File: GeeksCoreLibrary/Components/Configurator/Services/ConfiguratorsService.cs [hunk 1]
```diff
@@ -459,17 +459,22 @@ public async Task<VueConfiguratorDataModel> GetVueConfiguratorDataAsync(string n
...
         IF(
             templatesFromStepId.id IS NOT NULL AND templatesFromStepId.`value` NOT IN ('', '0'),
-            (SELECT CONCAT_WS('', `value`, long_value) FROM wiser_itemdetail WHERE item_id = templatesFromStepId.`value` AND `key` = 'values_template'),
+            (SELECT CONCAT_WS('', `value`, long_value) FROM wiser_itemdetail WHERE item_id = CONVERT(templatesFromStepId.`value`, UNSIGNED) AND `key` = 'values_template'),
             CONCAT_WS('', stepOptionTemplate.`value`, stepOptionTemplate.long_value)
         ) AS stepOptionTemplate,
         CONCAT_WS('', stepOptionsQuery.`value`, stepOptionsQuery.long_value) AS stepOptionsQuery,
         CONCAT_WS('', extraDataQuery.`value`, extraDataQuery.long_value) AS extraDataQuery,
         urlRegex.`value` AS urlRegex,
+        (SELECT JSON_OBJECTAGG(`key`, CONCAT_WS('', `value`, long_value)) FROM wiser_itemdetail WHERE item_id = step.id AND groupname = 'extra_data') AS extraData,
```
Comment (MarkVanDijkHG) [inline GeeksCoreLibrary/Components/Configurator/Services/ConfiguratorsService.cs:473]: Please use the constants in WiserTableNames.

File: GeeksCoreLibrary/Components/Configurator/Services/ConfiguratorsService.cs [hunk 2]
```diff
@@ -524,17 +533,22 @@ public async Task<VueConfiguratorDataModel> GetVueConfiguratorDataAsync(string n
         # Layout properties.
         IF(
             templatesFromStepId.id IS NOT NULL AND templatesFromStepId.`value` NOT IN ('', '0'),
-            (SELECT CONCAT_WS('', `value`, long_value) FROM wiser_itemdetail WHERE item_id = templatesFromStepId.`value` AND `key` = 'step_template'),
+            (SELECT CONCAT_WS('', `value`, long_value) FROM wiser_itemdetail WHERE item_id = CONVERT(templatesFromStepId.`value`, UNSIGNED) AND `key` = 'step_template'),
```
Comment (MarkVanDijkHG) [inline GeeksCoreLibrary/Components/Configurator/Services/ConfiguratorsService.cs:536]: Please use the constants in WiserTableNames.

File: GeeksCoreLibrary/Components/Configurator/Services/ConfiguratorsService.cs [hunk 3]
```diff
@@ -459,17 +459,22 @@ public async Task<VueConfiguratorDataModel> GetVueConfiguratorDataAsync(string n
         validationRegexErrorMessage.`value` AS validationRegexErrorMessage,
         IF(
             templatesFromStepId.id IS NOT NULL AND templatesFromStepId.`value` NOT IN ('', '0'),
-            (SELECT CONCAT_WS('', `value`, long_value) FROM wiser_itemdetail WHERE item_id = templatesFromStepId.`value` AND `key` = 'step_template'),
+            (SELECT CONCAT_WS('', `value`, long_value) FROM wiser_itemdetail WHERE item_id = CONVERT(templatesFromStepId.`value`, UNSIGNED) AND `key` = 'step_template'),
```
Comment (MarkVanDijkHG) [inline GeeksCoreLibrary/Components/Configurator/Services/ConfiguratorsService.cs:462]: Please use the constants in WiserTableNames.

File: GeeksCoreLibrary/Components/Configurator/Services/ConfiguratorsService.cs [hunk 4]
```diff
@@ -459,17 +459,22 @@ public async Task<VueConfiguratorDataModel> GetVueConfiguratorDataAsync(string n
...
             CONCAT_WS('', stepTemplate.`value`, stepTemplate.long_value)
         ) AS stepTemplate,
         IF(
             templatesFromStepId.id IS NOT NULL AND templatesFromStepId.`value` NOT IN ('', '0'),
-            (SELECT CONCAT_WS('', `value`, long_value) FROM wiser_itemdetail WHERE item_id = templatesFromStepId.`value` AND `key` = 'values_template'),
+            (SELECT CONCAT_WS('', `value`, long_value) FROM wiser_itemdetail WHERE item_id = CONVERT(templatesFromStepId.`value`, UNSIGNED) AND `key` = 'values_template'),
```
Comment (MarkVanDijkHG) [inline GeeksCoreLibrary/Components/Configurator/Services/ConfiguratorsService.cs:467]: Please use the constants in WiserTableNames.

File: GeeksCoreLibrary/Components/Configurator/Services/ConfiguratorsService.cs [hunk 5]
```diff
@@ -524,17 +533,22 @@ public async Task<VueConfiguratorDataModel> GetVueConfiguratorDataAsync(string n
...
```
Comment (MarkVanDijkHG) [inline GeeksCoreLibrary/Components/Configurator/Services/ConfiguratorsService.cs:541]: Please use the constants in WiserTableNames.

File: GeeksCoreLibrary/Components/Configurator/Services/ConfiguratorsService.cs [hunk 6]
```diff
@@ -524,17 +533,22 @@ public async Task<VueConfiguratorDataModel> GetVueConfiguratorDataAsync(string n
...
             CONCAT_WS('', stepOptionTemplate.`value`, stepOptionTemplate.long_value)
         ) AS stepOptionTemplate,
         CONCAT_WS('', stepOptionsQuery.`value`, stepOptionsQuery.long_value) AS stepOptionsQuery,
         CONCAT_WS('', extraDataQuery.`value`, extraDataQuery.long_value) AS extraDataQuery,
         urlRegex.`value` AS urlRegex,
+        (SELECT JSON_OBJECTAGG(`key`, CONCAT_WS('', `value`, long_value)) FROM wiser_itemdetail WHERE item_id = step.id AND groupname = 'extra_data') AS extraData,
```
Comment (MarkVanDijkHG) [inline GeeksCoreLibrary/Components/Configurator/Services/ConfiguratorsService.cs:547]: Please use the constants in WiserTableNames.

**Review by MarkVanDijkHG — CHANGES_REQUESTED** [review]

File: GeeksCoreLibrary/Components/Configurator/Services/ConfiguratorsService.cs (hunk 1 above)
Comment (HansJuice) [reply GeeksCoreLibrary/Components/Configurator/Services/ConfiguratorsService.cs]: Fixed.

File: GeeksCoreLibrary/Components/Configurator/Services/ConfiguratorsService.cs (hunk 2 above)
Comment (HansJuice) [reply GeeksCoreLibrary/Components/Configurator/Services/ConfiguratorsService.cs]: Fixed.

File: GeeksCoreLibrary/Components/Configurator/Services/ConfiguratorsService.cs (hunk 3 above)
Comment (HansJuice) [reply GeeksCoreLibrary/Components/Configurator/Services/ConfiguratorsService.cs]: Fixed.

File: GeeksCoreLibrary/Components/Configurator/Services/ConfiguratorsService.cs (hunk 4 above)
Comment (HansJuice) [reply GeeksCoreLibrary/Components/Configurator/Services/ConfiguratorsService.cs]: Fixed.

File: GeeksCoreLibrary/Components/Configurator/Services/ConfiguratorsService.cs (hunk 5 above)
Comment (HansJuice) [reply GeeksCoreLibrary/Components/Configurator/Services/ConfiguratorsService.cs]: Fixed.

File: GeeksCoreLibrary/Components/Configurator/Services/ConfiguratorsService.cs (hunk 6 above)
Comment (HansJuice) [reply GeeksCoreLibrary/Components/Configurator/Services/ConfiguratorsService.cs]: Fixed.

**Review by MarkVanDijkHG — APPROVED** [review]

**Pull request merged**